File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/file_read_configuration.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class FileReadConfiguration(object):
    """
    File read/upload settings collected by the UI and persisted as Agent Flow
    session configuration. DataHub stores and forwards these settings;
    runtime and gateway components enforce file policy during execution.
    """

    #: A constant which can be used with the allowed_file_types property of a FileReadConfiguration.
    #: This constant has a value of "pdf"
    ALLOWED_FILE_TYPES_PDF = "pdf"

    #: A constant which can be used with the allowed_file_types property of a FileReadConfiguration.
    #: This constant has a value of "txt"
    ALLOWED_FILE_TYPES_TXT = "txt"

    #: A constant which can be used with the allowed_file_types property of a FileReadConfiguration.
    #: This constant has a value of "csv"
    ALLOWED_FILE_TYPES_CSV = "csv"

    #: A constant which can be used with the allowed_file_types property of a FileReadConfiguration.
    #: This constant has a value of "json"
    ALLOWED_FILE_TYPES_JSON = "json"

    #: A constant which can be used with the allowed_file_types property of a FileReadConfiguration.
    #: This constant has a value of "ndjson"
    ALLOWED_FILE_TYPES_NDJSON = "ndjson"

    #: A constant which can be used with the allowed_file_types property of a FileReadConfiguration.
    #: This constant has a value of "png"
    ALLOWED_FILE_TYPES_PNG = "png"

    #: A constant which can be used with the allowed_file_types property of a FileReadConfiguration.
    #: This constant has a value of "jpg"
    ALLOWED_FILE_TYPES_JPG = "jpg"

    #: A constant which can be used with the allowed_file_types property of a FileReadConfiguration.
    #: This constant has a value of "jpeg"
    ALLOWED_FILE_TYPES_JPEG = "jpeg"

    def __init__(self, **kwargs):
        """
        Initializes a new FileReadConfiguration object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param is_enabled:
            The value to assign to the is_enabled property of this FileReadConfiguration.
        :type is_enabled: bool

        :param allowed_file_types:
            The value to assign to the allowed_file_types property of this FileReadConfiguration.
            Allowed values for items in this list are: "pdf", "txt", "csv", "json", "ndjson", "png", "jpg", "jpeg", 'UNKNOWN_ENUM_VALUE'.
            Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.
        :type allowed_file_types: list[str]

        :param max_file_size_mb:
            The value to assign to the max_file_size_mb property of this FileReadConfiguration.
        :type max_file_size_mb: int

        :param max_file_count:
            The value to assign to the max_file_count property of this FileReadConfiguration.
        :type max_file_count: int

        :param target_path:
            The value to assign to the target_path property of this FileReadConfiguration.
        :type target_path: str

        :param is_persistence_enabled:
            The value to assign to the is_persistence_enabled property of this FileReadConfiguration.
        :type is_persistence_enabled: bool

        :param retention_days:
            The value to assign to the retention_days property of this FileReadConfiguration.
        :type retention_days: int

        """
        self.swagger_types = {
            'is_enabled': 'bool',
            'allowed_file_types': 'list[str]',
            'max_file_size_mb': 'int',
            'max_file_count': 'int',
            'target_path': 'str',
            'is_persistence_enabled': 'bool',
            'retention_days': 'int'
        }

        self.attribute_map = {
            'is_enabled': 'isEnabled',
            'allowed_file_types': 'allowedFileTypes',
            'max_file_size_mb': 'maxFileSizeMb',
            'max_file_count': 'maxFileCount',
            'target_path': 'targetPath',
            'is_persistence_enabled': 'isPersistenceEnabled',
            'retention_days': 'retentionDays'
        }

        self._is_enabled = None
        self._allowed_file_types = None
        self._max_file_size_mb = None
        self._max_file_count = None
        self._target_path = None
        self._is_persistence_enabled = None
        self._retention_days = None

    @property
    def is_enabled(self):
        """
        Gets the is_enabled of this FileReadConfiguration.
        Whether file read/upload configuration is enabled.


        :return: The is_enabled of this FileReadConfiguration.
        :rtype: bool
        """
        return self._is_enabled

    @is_enabled.setter
    def is_enabled(self, is_enabled):
        """
        Sets the is_enabled of this FileReadConfiguration.
        Whether file read/upload configuration is enabled.


        :param is_enabled: The is_enabled of this FileReadConfiguration.
        :type: bool
        """
        self._is_enabled = is_enabled

    @property
    def allowed_file_types(self):
        """
        Gets the allowed_file_types of this FileReadConfiguration.
        File type enum values accepted by the UI/runtime contract.

        Allowed values for items in this list are: "pdf", "txt", "csv", "json", "ndjson", "png", "jpg", "jpeg", 'UNKNOWN_ENUM_VALUE'.
        Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.


        :return: The allowed_file_types of this FileReadConfiguration.
        :rtype: list[str]
        """
        return self._allowed_file_types

    @allowed_file_types.setter
    def allowed_file_types(self, allowed_file_types):
        """
        Sets the allowed_file_types of this FileReadConfiguration.
        File type enum values accepted by the UI/runtime contract.


        :param allowed_file_types: The allowed_file_types of this FileReadConfiguration.
        :type: list[str]
        """
        allowed_values = ["pdf", "txt", "csv", "json", "ndjson", "png", "jpg", "jpeg"]
        if allowed_file_types:
            allowed_file_types[:] = ['UNKNOWN_ENUM_VALUE' if not value_allowed_none_or_none_sentinel(x, allowed_values) else x for x in allowed_file_types]
        self._allowed_file_types = allowed_file_types

    @property
    def max_file_size_mb(self):
        """
        Gets the max_file_size_mb of this FileReadConfiguration.
        Maximum uploaded file size in megabytes.


        :return: The max_file_size_mb of this FileReadConfiguration.
        :rtype: int
        """
        return self._max_file_size_mb

    @max_file_size_mb.setter
    def max_file_size_mb(self, max_file_size_mb):
        """
        Sets the max_file_size_mb of this FileReadConfiguration.
        Maximum uploaded file size in megabytes.


        :param max_file_size_mb: The max_file_size_mb of this FileReadConfiguration.
        :type: int
        """
        self._max_file_size_mb = max_file_size_mb

    @property
    def max_file_count(self):
        """
        Gets the max_file_count of this FileReadConfiguration.
        Maximum number of files allowed in a request/session.


        :return: The max_file_count of this FileReadConfiguration.
        :rtype: int
        """
        return self._max_file_count

    @max_file_count.setter
    def max_file_count(self, max_file_count):
        """
        Sets the max_file_count of this FileReadConfiguration.
        Maximum number of files allowed in a request/session.


        :param max_file_count: The max_file_count of this FileReadConfiguration.
        :type: int
        """
        self._max_file_count = max_file_count

    @property
    def target_path(self):
        """
        Gets the target_path of this FileReadConfiguration.
        Strict relative workspace folder or volumeRef/workspaceFolder path where uploaded/readable files are stored or located for later runtime access. Path segments must not be empty, absolute, traversal segments, backslash-separated, contain control characters, or contain malformed dot-separated volume refs.


        :return: The target_path of this FileReadConfiguration.
        :rtype: str
        """
        return self._target_path

    @target_path.setter
    def target_path(self, target_path):
        """
        Sets the target_path of this FileReadConfiguration.
        Strict relative workspace folder or volumeRef/workspaceFolder path where uploaded/readable files are stored or located for later runtime access. Path segments must not be empty, absolute, traversal segments, backslash-separated, contain control characters, or contain malformed dot-separated volume refs.


        :param target_path: The target_path of this FileReadConfiguration.
        :type: str
        """
        self._target_path = target_path

    @property
    def is_persistence_enabled(self):
        """
        Gets the is_persistence_enabled of this FileReadConfiguration.
        Whether uploaded files should be retained beyond the current request/session intent.


        :return: The is_persistence_enabled of this FileReadConfiguration.
        :rtype: bool
        """
        return self._is_persistence_enabled

    @is_persistence_enabled.setter
    def is_persistence_enabled(self, is_persistence_enabled):
        """
        Sets the is_persistence_enabled of this FileReadConfiguration.
        Whether uploaded files should be retained beyond the current request/session intent.


        :param is_persistence_enabled: The is_persistence_enabled of this FileReadConfiguration.
        :type: bool
        """
        self._is_persistence_enabled = is_persistence_enabled

    @property
    def retention_days(self):
        """
        Gets the retention_days of this FileReadConfiguration.
        Retention period in days for persisted files.


        :return: The retention_days of this FileReadConfiguration.
        :rtype: int
        """
        return self._retention_days

    @retention_days.setter
    def retention_days(self, retention_days):
        """
        Sets the retention_days of this FileReadConfiguration.
        Retention period in days for persisted files.


        :param retention_days: The retention_days of this FileReadConfiguration.
        :type: int
        """
        self._retention_days = retention_days

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/FileReadConfiguration.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * File read/upload settings collected by the UI and persisted as Agent Flow
* session configuration. DataHub stores and forwards these settings;
* runtime and gateway components enforce file policy during execution.
* 
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=FileReadConfiguration.Builder.class)

public final class FileReadConfiguration  {
    @Deprecated
    @java.beans.ConstructorProperties({"isEnabled", "allowedFileTypes", "maxFileSizeMb", "maxFileCount", "targetPath", "isPersistenceEnabled", "retentionDays"})
    public FileReadConfiguration(Boolean isEnabled, java.util.List<FileAllowedFileType> allowedFileTypes, Long maxFileSizeMb, Integer maxFileCount, String targetPath, Boolean isPersistenceEnabled, Integer retentionDays) {
        super();
        this.isEnabled = isEnabled;
        this.allowedFileTypes = allowedFileTypes;
        this.maxFileSizeMb = maxFileSizeMb;
        this.maxFileCount = maxFileCount;
        this.targetPath = targetPath;
        this.isPersistenceEnabled = isPersistenceEnabled;
        this.retentionDays = retentionDays;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Whether file read/upload configuration is enabled.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("isEnabled")
private Boolean isEnabled;

        /**
         * Whether file read/upload configuration is enabled.
         * @param isEnabled the value to set
         * @return this builder
         **/
        

public Builder isEnabled(Boolean isEnabled) {
    this.isEnabled = isEnabled;
    return this;
}
            /**
     * File type enum values accepted by the UI/runtime contract.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("allowedFileTypes")
private java.util.List<FileAllowedFileType> allowedFileTypes;

        /**
         * File type enum values accepted by the UI/runtime contract.
         * @param allowedFileTypes the value to set
         * @return this builder
         **/
        

public Builder allowedFileTypes(java.util.List<FileAllowedFileType> allowedFileTypes) {
    this.allowedFileTypes = allowedFileTypes;
    return this;
}
            /**
     * Requested maximum uploaded file size in megabytes. When omitted, Agent Service applies the 10 MiB default; explicit values may raise the per-file limit up to 50 MiB while the active session remains capped at 500 MiB.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("maxFileSizeMb")
private Long maxFileSizeMb;

        /**
         * Requested maximum uploaded file size in megabytes. When omitted, Agent Service applies the 10 MiB default; explicit values may raise the per-file limit up to 50 MiB while the active session remains capped at 500 MiB.
         * @param maxFileSizeMb the value to set
         * @return this builder
         **/
        

public Builder maxFileSizeMb(Long maxFileSizeMb) {
    this.maxFileSizeMb = maxFileSizeMb;
    return this;
}
            /**
     * Maximum number of active files allowed in a session. When omitted, Agent Service applies the five-file default; explicit values may raise the limit to twenty while the aggregate byte quota remains authoritative.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("maxFileCount")
private Integer maxFileCount;

        /**
         * Maximum number of active files allowed in a session. When omitted, Agent Service applies the five-file default; explicit values may raise the limit to twenty while the aggregate byte quota remains authoritative.
         * @param maxFileCount the value to set
         * @return this builder
         **/
        

public Builder maxFileCount(Integer maxFileCount) {
    this.maxFileCount = maxFileCount;
    return this;
}
            /**
     * Canonical /Volumes/catalog/schema/volume path, optionally followed by safe folder segments, or a legacy relative volumeRef/workspaceFolder path where uploaded/readable files are stored or located for later runtime access.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("targetPath")
private String targetPath;

        /**
         * Canonical /Volumes/catalog/schema/volume path, optionally followed by safe folder segments, or a legacy relative volumeRef/workspaceFolder path where uploaded/readable files are stored or located for later runtime access.
         * @param targetPath the value to set
         * @return this builder
         **/
        

public Builder targetPath(String targetPath) {
    this.targetPath = targetPath;
    return this;
}
            /**
     * Whether uploaded files should be retained beyond the current request/session intent.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("isPersistenceEnabled")
private Boolean isPersistenceEnabled;

        /**
         * Whether uploaded files should be retained beyond the current request/session intent.
         * @param isPersistenceEnabled the value to set
         * @return this builder
         **/
        

public Builder isPersistenceEnabled(Boolean isPersistenceEnabled) {
    this.isPersistenceEnabled = isPersistenceEnabled;
    return this;
}
            /**
     * Retention period in days for persisted files.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("retentionDays")
private Integer retentionDays;

        /**
         * Retention period in days for persisted files.
         * @param retentionDays the value to set
         * @return this builder
         **/
        

public Builder retentionDays(Integer retentionDays) {
    this.retentionDays = retentionDays;
    return this;
}


        public FileReadConfiguration build() {
            FileReadConfiguration model = new FileReadConfiguration(this.isEnabled
                , this.allowedFileTypes
                , this.maxFileSizeMb
                , this.maxFileCount
                , this.targetPath
                , this.isPersistenceEnabled
                , this.retentionDays);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(FileReadConfiguration model) {
                this.isEnabled(model.getIsEnabled());
    this.allowedFileTypes(model.getAllowedFileTypes());
    this.maxFileSizeMb(model.getMaxFileSizeMb());
    this.maxFileCount(model.getMaxFileCount());
    this.targetPath(model.getTargetPath());
    this.isPersistenceEnabled(model.getIsPersistenceEnabled());
    this.retentionDays(model.getRetentionDays());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Whether file read/upload configuration is enabled.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("isEnabled")
    private final Boolean isEnabled;

        /**
     * Whether file read/upload configuration is enabled.
     * @return the value
     **/
    
    public Boolean getIsEnabled() {
        return isEnabled;
    }

    
        /**
     * File type enum values accepted by the UI/runtime contract.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("allowedFileTypes")
    private final java.util.List<FileAllowedFileType> allowedFileTypes;

        /**
     * File type enum values accepted by the UI/runtime contract.
     * @return the value
     **/
    
    public java.util.List<FileAllowedFileType> getAllowedFileTypes() {
        return allowedFileTypes;
    }


        /**
     * Requested maximum uploaded file size in megabytes. When omitted, Agent Service applies the 10 MiB default; explicit values may raise the per-file limit up to 50 MiB while the active session remains capped at 500 MiB.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("maxFileSizeMb")
    private final Long maxFileSizeMb;

        /**
     * Requested maximum uploaded file size in megabytes. When omitted, Agent Service applies the 10 MiB default; explicit values may raise the per-file limit up to 50 MiB while the active session remains capped at 500 MiB.
     * @return the value
     **/
    
    public Long getMaxFileSizeMb() {
        return maxFileSizeMb;
    }


        /**
     * Maximum number of active files allowed in a session. When omitted, Agent Service applies the five-file default; explicit values may raise the limit to twenty while the aggregate byte quota remains authoritative.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("maxFileCount")
    private final Integer maxFileCount;

        /**
     * Maximum number of active files allowed in a session. When omitted, Agent Service applies the five-file default; explicit values may raise the limit to twenty while the aggregate byte quota remains authoritative.
     * @return the value
     **/
    
    public Integer getMaxFileCount() {
        return maxFileCount;
    }


        /**
     * Canonical /Volumes/catalog/schema/volume path, optionally followed by safe folder segments, or a legacy relative volumeRef/workspaceFolder path where uploaded/readable files are stored or located for later runtime access.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("targetPath")
    private final String targetPath;

        /**
     * Canonical /Volumes/catalog/schema/volume path, optionally followed by safe folder segments, or a legacy relative volumeRef/workspaceFolder path where uploaded/readable files are stored or located for later runtime access.
     * @return the value
     **/
    
    public String getTargetPath() {
        return targetPath;
    }


        /**
     * Whether uploaded files should be retained beyond the current request/session intent.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("isPersistenceEnabled")
    private final Boolean isPersistenceEnabled;

        /**
     * Whether uploaded files should be retained beyond the current request/session intent.
     * @return the value
     **/
    
    public Boolean getIsPersistenceEnabled() {
        return isPersistenceEnabled;
    }


        /**
     * Retention period in days for persisted files.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("retentionDays")
    private final Integer retentionDays;

        /**
     * Retention period in days for persisted files.
     * @return the value
     **/
    
    public Integer getRetentionDays() {
        return retentionDays;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("FileReadConfiguration(");
        sb.append("isEnabled=").append(String.valueOf(this.isEnabled));
        sb.append(", allowedFileTypes=").append(String.valueOf(this.allowedFileTypes));
        sb.append(", maxFileSizeMb=").append(String.valueOf(this.maxFileSizeMb));
        sb.append(", maxFileCount=").append(String.valueOf(this.maxFileCount));
        sb.append(", targetPath=").append(String.valueOf(this.targetPath));
        sb.append(", isPersistenceEnabled=").append(String.valueOf(this.isPersistenceEnabled));
        sb.append(", retentionDays=").append(String.valueOf(this.retentionDays));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof FileReadConfiguration)) {
            return false;
        }

        FileReadConfiguration other = (FileReadConfiguration) o;
        return java.util.Objects.equals(this.isEnabled, other.isEnabled) &&
            java.util.Objects.equals(this.allowedFileTypes, other.allowedFileTypes) &&
            java.util.Objects.equals(this.maxFileSizeMb, other.maxFileSizeMb) &&
            java.util.Objects.equals(this.maxFileCount, other.maxFileCount) &&
            java.util.Objects.equals(this.targetPath, other.targetPath) &&
            java.util.Objects.equals(this.isPersistenceEnabled, other.isPersistenceEnabled) &&
            java.util.Objects.equals(this.retentionDays, other.retentionDays);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.isEnabled == null ? 43 : this.isEnabled.hashCode());
        result = (result * PRIME) + (this.allowedFileTypes == null ? 43 : this.allowedFileTypes.hashCode());
        result = (result * PRIME) + (this.maxFileSizeMb == null ? 43 : this.maxFileSizeMb.hashCode());
        result = (result * PRIME) + (this.maxFileCount == null ? 43 : this.maxFileCount.hashCode());
        result = (result * PRIME) + (this.targetPath == null ? 43 : this.targetPath.hashCode());
        result = (result * PRIME) + (this.isPersistenceEnabled == null ? 43 : this.isPersistenceEnabled.hashCode());
        result = (result * PRIME) + (this.retentionDays == null ? 43 : this.retentionDays.hashCode());
        return result;
    }


}
